api/v3/Contribution.php

```diff
@@ -579,6 +579,8 @@ function _civicrm_api3_contribution_completetransaction_spec(&$params) {
 function civicrm_api3_contribution_repeattransaction($params) {
   $input = $ids = [];
   civicrm_api3_verify_one_mandatory($params, NULL, ['contribution_recur_id', 'original_contribution_id']);
+  // @todo Clean this up - we should pass through either original_contribution_id or contribution_recur_id
```

Comment:
Once we get rid of the other place that calls completeOrder none of this makes sense inn the api class & it all looks a lot different once you consolidate it in the other function

Reply:
Agree with that as the goal - but we have over 100 lines of code to work through before we can get there with various bugs (such as the template contribution issue) to work through. I don't see a way to get there without simplifying some of this stuff first.